torch/distributed/_mesh_layout.py

```diff
@@ -243,3 +244,90 @@ def check_non_overlap(self) -> bool:
         """
         ranks = self.all_ranks_from_zero()
         return len(ranks) == len(set(ranks))
+
+    def remap_to_tensor(
+        self,
+        mesh_tensor: torch.Tensor,
+        world_size: int,
```

Comment:
If 1) we don't need all ranks to participate in sub pg creation, 2) all operation, including flatten and slicing are functional, can I claim that world_size should be needed for this function because both mesh_tensor and layout will be scaled appropriately. So world_size is always the same as mesh_tensor.numel() and we don't need to worry about sub PG creation.

Reply:
If my new understanding is correct, then my original comment was incorrect that this `remap_to_tensor()` can be a standalone PR because its argument is tightly coupled with DeviceMesh and PG details. Then we may need to describe the restriction (sub PG creation and slicing is not functional)